src/main/java/com/google/codeu/data/Message.java

```diff
@@ -26,21 +26,23 @@
   private String text;
   private long timestamp;
   private String recipient;
+  private float sentimentScore;
 
   /**
    * Constructs a new {@link Message} posted by {@code user} with {@code text} content and {@code recipient}. Generates a
    * random ID and uses the current system time for the creation time.
    */
-  public Message(String user, String text, String recipient) {
-    this(UUID.randomUUID(), user, text, System.currentTimeMillis(), recipient);
+  public Message(String user, String text, String recipient, float sentimentScore) {
+    this(UUID.randomUUID(), user, text, System.currentTimeMillis(), recipient, sentimentScore);
   }
 
-  public Message(UUID id, String user, String text, long timestamp, String recipient) {
+  public Message(UUID id, String user, String text, long timestamp, String recipient, float sentimentScore) {
```

Comment:
Since the sentiment score isn't really intrinsic to the Message but is a derived computation about the message, it might be a good idea to create a separate entity and object for it. That is, I would recommend thinking about it/structuring it as a `MessageSentimentScore` references a Message (by key) and contains the computed score (as well as the time at which the sentiment was calculated) and acts like a cache to make it quicker to obtain the score.

For example, if you create another endpoint to obtain the sentiment score of a message, it could act like:

       if exists already computed score for message key:
          return existing score
       compute sentiment score
       store score for future efficiency
       return computed score

There are probably more places where you need just the message content, though, without the score. So it may not make sense to tie the two so tightly.